src/hooks/useTableManagement.ts

```
@@ -0,0 +1,183 @@
+import type { UsageData } from "@/types/table";
+import { useState, useCallback, useEffect } from "react";
+import { useSearchParams } from "react-router-dom";
+import { toast } from "react-toastify";;
+
+interface UseTableManagementProps {
+  fetchFn: (params: { page: number; size: number }) => Promise<{
+    pass: boolean;
+    data: {
+      [key: string]: any;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)